src/rust/engine/dep_inference/src/python/visitor.rs

```diff
@@ -0,0 +1,437 @@
+// Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+#[derive(Debug, PartialEq)]
+pub enum ChildBehavior {
+  Visit,
+  Ignore,
+}
```

Comment:
It is. What's the question?
Are you asking about the behavior of visiting? This is really useful for user-implemented visits. E.g. "I've done the traversing, you don't need to do it".

The `visit_try_statement` returns `Ignore`, for instance.

Reply:
Something that is generated by `build.rs` shouldn't need to be committed to the repo: it will be regenerated when necessary.

Maybe you're not generating it inside of the relevant output directory? https://doc.rust-lang.org/cargo/reference/build-script-examples.html#code-generation refers to using the `OUT_DIR` environment variable.

Reply:
AIUI, there's a bit of a school of thought of not doing simple codegen in `build.rs` as it forces every build to spend time compiling and run the build script, and it can be better to just generate it to disk, rather than spend the time generating the exact same file on every build.

That said, I think this is particularly for published libraries, where codegen-in-build.rs it's a cost for the whole ecosystem, not just a single project like this (since every reserve dependency build also has to compile and run the build script).

Reply:
We already generated code from `build.rs` as part of protobuf compilation, and afaict, we've never had a correctness issue there.